lib/ansible/modules/cloud/amazon/emr_cluster_facts.py

```diff
@@ -0,0 +1,317 @@
+          description: The list of Configurations supplied to the EMR cluster
+          returned: always
+          type: list
+          sample: []
+        security_configuration:
+          description: The name of the security configuration applied to the cluster
+          returned: always
+          type: string
+          sample: ''
+        auto_scaling_role:
+          description: An IAM role for automatic scaling policies
+          returned: always
+          type: string
+          sample: 'AwesomeScalingRole'
+        scale_down_behavior:
+          description: The way that individual Amazon EC2 instances terminate when an automatic scale-in activity occurs
+          returned: always
+          type: string
+          sample: 'TERMINATE_AT_TASK_COMPLETION'
+        custom_ami_id:
+          description: The ID of a custom Amazon EBS-backed Linux AMI
+          returned: always
+          type: string
+          sample: 'ami-1234567'
+        ebs_root_volume_size:
+          description: The size, in GiB, of the EBS root device volume of the Linux AMI that is used for each EC2 instance
+          returned: always
+          type: integer
+          sample: 25
+        repo_upgrade_on_boot:
+          description: Specifies the type of updates that are applied from the Amazon Linux AMI package repositories
+          returned: always
+          type: string
+          sample: ''
+'''
+
+
+from collections import defaultdict
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+
+class EMRConnection(object):
+
+    def __init__(self, module, region, **aws_connect_params):
+        try:
+            self.connection = boto3_conn(module, conn_type='client',
+                                         resource='emr', region=region,
+                                         **aws_connect_params)
+            self.module = module
+        except Exception as e:
+            module.fail_json(msg="Failed to connect to AWS: %s" % str(e))
+        self.region = region
+
+    def get_emr_cluster(self, cluster_id=None):
+        try:
+            results = self.connection.describe_cluster(ClusterId=cluster_id)
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self.module.fail_json_aws(e, msg="Couldn't get EMR cluster facts")
+        return camel_dict_to_snake_dict(results)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        id=dict()
+    ))
+
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=True)
+
+    region, dummy, aws_connect_params = get_aws_connection_info(module, boto3=True)
```

Comment:
Since you're using AnsibleAWSModule, you don't need to call get_aws_connection_info. You can do `connection = module.client('emr')` (no exception handling needed around that either). You can pass connection and module to get_emr_cluster in addition to `cluster_id=None`.

Reply:
Hmm, I wonder if I'm implementing this wrong.  When I try to set up the connection this way, I get an error telling me that `module` doesn't have the `client` attribute.  I've looked at other modules that do it this way and it works but it's not working for me in this case.  I'm not sure what I'm doing differently that would cause this behaviour.